src/mlang/backend_ir/bir_interpreter.ml

```diff
@@ -571,6 +571,7 @@ module Make (N : Bir_number.NumberInterface) = struct
             in
             let maxi = ref (access_index 0) in
             for i = 0 to Int64.to_int up do
+              (* Fragile: rely on polymorphic compare where None < Some _ *)
```

Comment:
Can you explicit this comparison with a match? This is the place where the reference semantics is so we want it to be very explicit.